esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -2051,6 +2061,20 @@ void WsWuInfo::getWorkunitThorSlaveLog(IPropertyTree* directories, const char *p
     getWorkunitThorSlaveLog(nodeGroup, ipAddress, logDate, logDir.str(), slaveNum, buf, outFile, forDownload);
 }
 
+void WsWuInfo::getWorkunitThorSlaveLogSingleFile(const char* thorProcess, const char* logDate, int slaveNum,
+    MemoryBuffer& buf, const char* outFile)
+{
+    StringBuffer logPath;
+    getWorkunitProcessLogPath("Thor", logPath);
+
+    VStringBuffer logFileName("%s%c%s_thorslave.%u.%s.log", logPath.str(), PATHSEPCHAR, thorProcess, slaveNum, logDate);
```

Comment:
How about adding 'thorSlaveLogSearchString=_thorslave' under Debug/imported.

Reply:
Relating to comments elsewhere, my feeling is that this would be cleaner if all pertanent info. was captured in the workunit under Process/ including whether @singleLog.

You could then expect IConstWorkUnit::getLogs() to retrieve any log, including slave logs.
But you might want to extend it, so it can give you a specific slave log, e.g:
```
IStringIterator *CLocalWorkUnit::getLogs(const char *type, const char *instance, unsigned n) const
{
     // See if Process/<type>/<instance> has a @pattern .. If has @pattern, should also have @max (for # slaves)
     // if has pattern and n>0, construct log using pattern and n. Error if n>max.
     // if has pattern and n undefined/default, construct logs for all up to @max.
}
```

```
Owned<IStringIterator> logs = workunit->getLogs("Thor", "mythor", 5);
```

Reply:
Also, the introduction of IConstWUThorLogInfo may be sensible, but it's questionable why there is now 2 different ways to get log info getWUThorLogInfo() and getLogs()

I suspect you can simplify, so that everything uses getWUThorLogInfo ?

Reply:
And I think when simplified, your code WsWuInfo::getWorkunitThorSlaveLogSingleFile code may go away.
With everything universally using IConstWUThorLogInfo (which will encode whether @snigle or not too), then your code based on IConstWUThorLogInfo::isSingleLog(), calling your getWorkunitLogSingleFile

Reply:
Will check